autoload/lsp/symbol.vim

```diff
@@ -291,7 +304,13 @@ export def GotoSymbol(lspserver: dict<any>, location: dict<any>, peekSymbol: boo
     # jump to the file and line containing the symbol
     var wid = fname->bufwinid()
     if wid != -1
-      wid->win_gotoid()
+      # `wid` maybe just is one of windows in cur tab which had same buf
+      # jump to `wid` only if cur one is not same buf or maybe a bit mess
+      # which should always try to re-use cur one firstly (if it is same)
+      # otherwise cursor perhaps dumbly jumpped to `wid` though same buf
+      if bufwinid(bufnr()) != wid
```

Comment:
This condition is true if the current window ID is not the same as the window ID of the buffer containing fname.  This doesn't match the description in the comment.

Reply:
there should be another 
```vim
        # if had one in cur tab
        # and cur win is same buf
        if bufwinid(bufnr()) == wid
          # do nothing
        else
          # or pick up one in cur tab
          wid->win_gotoid()
        endif
```

i simplified it just `bufwinid(bufnr()) != wid`